packages/models-library/src/models_library/clusters.py

```diff
@@ -102,6 +120,10 @@ class BaseCluster(BaseModel):
     )
     access_rights: dict[GroupID, ClusterAccessRights] = Field(default_factory=dict)
 
+    _from_equivalent_enums = validator("type", allow_reuse=True, pre=True)(
+        create_enums_pre_validator(ClusterTypeInModel)
+    )
```

Comment:
If I get it, this is a validator. Why is it register like this and not as a decorated function?
I find it less readable.

Reply:
This technique is used to reuse validators. The validators I add in `models_library.utils.common_validators` are solutions to common pitfalls. The Enum problem is one of them and therefore it is very convenient to have a tested solution in place instead of being reinventing a solution in place.

Reply:
I got that.
My question was more in the likes of why not do something like this:

```python
@validator(allow_reuse=True, pre=True)
@classmethod
def _from_equivalent_enums(cli, values):
    return create_enums_pre_validator(values)
```

Since this is the format to whom we are normally used to?